src/Functions/FunctionsHashing.h

```diff
@@ -190,6 +192,33 @@ T combineHashesFunc(T t1, T t2)
     return HashFunction::apply(reinterpret_cast<const char *>(hashes), sizeof(hashes));
 }
 
+struct RipeMD160Impl
```

Comment:
If I understand correctly we use here a function from OpenSSL, but the struct is not under `#if USE_SSL`, so the build without it will fail. You can look at `HalfMD5Impl` for example.

`using FunctionRipeMD160Hash = FunctionAnyHash<RipeMD160Impl>;` would also need to be under `#if USE_SSL`

Reply:
i agree. however, the current edits doesn't take into account that `factory.registerFunction<FunctionRipeMD160Hash>(...)` would also need to be under #if USE_SSL. i will fix that today